### ISSUE TITLE
Import database through Excel fail

### ISSUE THREAD

**Issue by MaximeAgez** [LCA-ActivityBrowser/activity-browser#613]
Hi!

I tried sharing a database through the Excel import function and AB simply crashed (see the screenshot below). I'm on the dev version of AB, which I updated an hour ago.

![image](https://user-images.githubusercontent.com/24285636/123688592-31bdaf80-d820-11eb-9795-279185bbde18.png)

The database itself is somewhat big but nothing too crazy (80 processes and about the same amount of parameters).
There is no error message in the command window. It's just saying that there are some unlinked exchanges and then proceeds to list 10 of them (see below).

![image](https://user-images.githubusercontent.com/24285636/123688771-6b8eb600-d820-11eb-94a2-1e50c5296c3b.png)

Right now it seems complicated, maybe even impossible, to exchange fully operating brightway2 databases with colleagues. As pointed in this issue, Excel import crashes (although it might not be the case for everyone) and the bw2package import has issues with the parameters export as pointed in issue #579.

Thanks!

**Comment by marc-vdm** [comment]
Thanks for sharing this.

Which version are you currently using? We have recently made small changes to database imports and how processes can reference to other processes. I think these changes are currently still only available in the -dev branch.

Would it be possible for you to share the database file (considering confidentiality) too? 
If not, could you make the database smaller and keep a broken process in, and then perhaps change the names/numbers to keep data confidential?

The file could help in understanding where the issue lies.

**Comment by MaximeAgez** [comment]
I'm using the dev version of AB (same version on both machines). Here is a screenshot with the config of the environment (Python v3.8.10):

![image](https://user-images.githubusercontent.com/24285636/124293011-33d78500-db24-11eb-927e-14d99746913e.png)

and here a shortened version of the Excel file used for the import:

[export.xlsx](https://github.com/LCA-ActivityBrowser/activity-browser/files/6755370/export.xlsx)

Names of the database, i.e., "ecoinvent3.6 cut-off" are the same in both version of the ABs (the one exporting and the one importing). The only difference is the name of the brightway2 project.

**Comment by marc-vdm** [comment]
The database file you shared imports without issue for me on both the latest dev version and on my (slightly older) stable version of AB. 

Could you confirm the database file you shared is still problematic for you?
If it does, please update to the latest dev version and let us know if the issue persists :)

**Comment by MaximeAgez** [comment]
Apparently there might be a problem within the Excel file (I can't find what though). Some colleague told me they could read the shortened version I just shared, but not the full one. So here is the full one:

[export.xlsx](https://github.com/LCA-ActivityBrowser/activity-browser/files/6757259/export.xlsx)

**Comment by marc-vdm** [comment]
I was able to progress a little, but not fix your problem unfortunately.
There are three exchanges in the file that do not exist. These are:
line 535 'diesel, burned in mobile unit'
line 553 'automate inflation tyrebox, production'
line 563 'tyrebox for truck, production'
Also note that the parameter on line 15 'distance_saved' is used only on line 535.

You may want to add the processes, or remove the exchanges from the database. However, I tried that, and unfortunately, the file still fails to import.
If you want the file, you can find it here: [export_db_test_fix.xlsx](https://github.com/LCA-ActivityBrowser/activity-browser/files/6762377/export_db_test_fix.xlsx)

@StpdFox  could you look at the error below and see if I'm missing anything?

```
Extracted 1 worksheets in 0.97 seconds
Applying strategy: alter_database_name
Applying strategy: csv_restore_tuples
Applying strategy: csv_restore_booleans
Applying strategy: csv_numerize
Applying strategy: csv_drop_unknown
Applying strategy: csv_add_missing_exchanges_section
Applying strategy: csv_rewrite_product_key
Applying strategy: normalize_units
Applying strategy: normalize_biosphere_categories
Applying strategy: normalize_biosphere_names
Applying strategy: strip_biosphere_exc_locations
Applying strategy: set_code_by_activity_hash
Applying strategy: link_iterable_by_fields
Applying strategy: assign_only_product_as_production
Applying strategy: link_technosphere_by_activity_hash
Applying strategy: drop_falsey_uncertainty_fields_but_keep_zeros
Applying strategy: convert_uncertainty_types_to_integers
Applying strategy: hash_parameter_group
Applying strategy: convert_activity_parameters_to_list
Applied 19 strategies in 0.22 seconds
Applying strategy: relink_exchanges_with_db
Writing activities to SQLite3 database:
0% [##############################] 100% | ETA: 00:00:00
Total time elapsed: 00:00:00
Title: Writing activities to SQLite3 database:
  Started: 07/04/2021 18:38:08
  Finished: 07/04/2021 18:38:08
  Total time elapsed: 00:00:00
  CPU %: 33.60
  Memory %: 3.47
Traceback (most recent call last):
  File "C:\Users\m\miniconda3\envs\ab_dev\lib\site-packages\activity_browser\ui\wizards\db_import_wizard.py", line 698, in run
    self.run_local_import()
  File "C:\Users\m\miniconda3\envs\ab_dev\lib\site-packages\activity_browser\ui\wizards\db_import_wizard.py", line 817, in run_local_import
    self.archive_path, self.db_name, self.relink
  File "C:\Users\m\miniconda3\envs\ab_dev\lib\site-packages\activity_browser\bwutils\importers.py", line 107, in simple_automated_import
    db = obj.write_database(delete_existing=True, activate_parameters=True)
  File "C:\Users\m\miniconda3\envs\ab_dev\lib\site-packages\activity_browser\bwutils\importers.py", line 39, in write_database
    return super(ExcelImporter, self).write_database(**kwargs)
  File "C:\Users\m\miniconda3\envs\ab_dev\lib\site-packages\bw2io\importers\base_lci.py", line 269, in write_database
    db.write(existing)
  File "C:\Users\m\miniconda3\envs\ab_dev\lib\site-packages\bw2data\project.py", line 356, in writable_project
    return wrapped(*args, **kwargs)
  File "C:\Users\m\miniconda3\envs\ab_dev\lib\site-packages\bw2data\backends\peewee\database.py", line 266, in write
    self.make_searchable(reset=True)
  File "C:\Users\m\miniconda3\envs\ab_dev\lib\site-packages\bw2data\project.py", line 356, in writable_project
    return wrapped(*args, **kwargs)
  File "C:\Users\m\miniconda3\envs\ab_dev\lib\site-packages\bw2data\backends\peewee\database.py", line 311, in make_searchable
    IndexManager(self.filename).add_datasets(self)
  File "C:\Users\m\miniconda3\envs\ab_dev\lib\site-packages\bw2data\search\indices.py", line 47, in add_datasets
    writer.add_document(**self._format_dataset(ds))
  File "C:\Users\m\miniconda3\envs\ab_dev\lib\site-packages\bw2data\search\indices.py", line 33, in _format_dataset
    comment=ds.get(u"comment", u"").lower(),
AttributeError: 'NoneType' object has no attribute 'lower'


Database import interrupted!
```

**Comment by MaximeAgez** [comment]
I had the same error indeed. Probably related to the missing activities, which I had removed initially to test if copied activities were the source of the problems and forgot to put them back.

I tried on my own computer by creating a new project and the import worked just fine (with all activities). So the issue lies with the computer on which the import initially failed. I don't know what it is but I might need to reboot it because other things unrelated to AB seems funky. In the end there is no problem with the Excel import, my bad :)

Thanks for your help @marc-vdm
